Improved project setup, added some MC integration

xtphty opened the pull request.

xtphty committed:
Fixed issues in pom, added testing tools, fixed ignores for eclipse

xtphty committed:
Added whitelist grace period and registration command, tested using spigot 1.88

PlayerListener will run an async task on bukkit, after 60 seconds (configurable) it kicks the user

RegisterCommand is where we can add a post request to dgg to verify the private key (first agrument)

Tested using local spigot server (1.8.8):

[00:20:43 INFO]: [SubWhitelister] Register: username: xtphty , uuid: 43685a5f-b5e6-4dfe-a278-fb768575116e, authToken: auth-token, privateKey: hello
[00:21:38 INFO]: xtphty lost connection: Authentication with destiny.gg failed

hayksaakian commented:
As i understand it: Private key is basically a key held on the minecraft server, authtoken is a users destiny gg api key. 

If you send both with a uuid you can know that an exact minecraft user is requesting permission to a valid minecraft server, and with the apikey you can validate their sub level.

sztanpet commented:
correct, the private key is purely to protect the dgg api from "abuse"

sztanpet commented:
also, to give you a hint about using the api, ask me in private for the privatekey and authtokens for users with and without a sub

```
THIS IS A GET:
curl -v "stage.destiny.gg/auth/minecraft?privatekey=ccccccccccccccccccc&uuid=bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb"
THIS IS A POST:
curl -v --data "privatekey=ccccccccccccccccccc&authtoken=aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa&uuid=bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb" stage.destiny.gg/auth/minecraft
```

sztanpet commented on src/main/java/net/bitjump/bukkit/subwhitelister/commands/RegisterCommand.java:
```diff
@@ -0,0 +1,56 @@
+ *
+ */
+public class RegisterCommand implements CommandExecutor {
+	private static final String USAGE = "Invalid command, use: /register <private key>";
```
this is what we are talking about by the way, it should say `<login key from destiny.gg>` by default and make it configurable from the config
the private key is also a configurable, never exposed thing that is simply sent along with everything to prove that the minecraft server is the one doing the api request
so the authToken is what we are asking the user for, and it is exposed as "login key" in the d.gg profile

xtphty committed:
Added destinygg auth service and player listener

Auth service is done, a couple possible todos left on dgg api side.

Need to figure out the best way to whitelist players with an async task,
kick can work but needs to be done in a sync thread.

xtphty commented:
Can I get admin on stage.dgg? want to test this out tomorrow using a local server will be nice if i can give / take sub away from people.

sztanpet commented:
no such feature

sztanpet committed:
update readme

xtphty committed:
Cleanup and improved API calls and response parsing

Removed all the old mostly unused mod references.

Re-worked the API requests using a better rest client.

xtphty committed:
Added some more abstraction to improve configuration options

The config file now allows specifying a listener class implementation to
use for the whitelist process

Extracted some implementation details out of destiny.gg specific classes

xtphty committed:
Merged with branch 'xtphty/master'
